pkg/controllers/manila/manila.go

```diff
@@ -96,26 +94,10 @@ func (c *ManilaController) sync(ctx context.Context, syncCtx factory.SyncContext
 		return nil
 	}
 
-	var err403 gophercloud.ErrDefault403
-	var errNoEndpoint *gophercloud.ErrEndpointNotFound
 	shareTypes, err := c.openStackClient.GetShareTypes()
 	if err != nil {
-		switch {
-		case errors.As(err, &err403):
-			// User doesn't have permissions to list share types, report the operator as disabled
-			klog.V(4).Infof("User doesn't have access to Manila service: %v", err)
-			return c.setDisabledCondition("User doesn't have access to Manila service")
-		case isNotFoundError(err):
-			// Manila Share Type API is not available
-			klog.V(4).Infof("Cannot find API to fetch Manila share types: %v", err)
-			return c.setDisabledCondition("Cannot find API to fetch Manila share types")
-		case errors.As(err, &errNoEndpoint):
-			// OpenStack does not support manila, report the operator as disabled
-			klog.V(4).Infof("This OpenStack cluster does not provide Manila service: %v", err)
-			return c.setDisabledCondition("This OpenStack cluster does not provide Manila service")
-		default:
-			return err
-		}
+		klog.V(4).Infof("Unable to retrieve Manila share types: %v", err)
+		return c.setDisabledCondition("Unable to retrieve Manila share types")
```

Comment:
Can you add the error message here, so that it shows up in the CR? Also, this would remove the need to log the message in the line above IMO.

Reply:
Can the message in the OperatorCondition be anything? If yes, I don't have objection adding the error message to the CR.
I'm little less convinced about removing the log message.

Reply:
The message in the operator condition is a generic string, it can be anything.

Reply:
>  I'm little less convinced about removing the log message.

Usually I prefer to handle the error once, but if you really want to log in addition to reporting the error via via CR, I'd use `Errorf` instead.

Reply:
I've finally removed the log message as `Errorf` was not recognized as valid by my editor and didn't want to introduce a regression.